app/controllers/application_templates_controller.rb

```diff
@@ -27,7 +27,24 @@ def toggle_active
   private
 
   def find_template
-    params.require :id
+    if params[:specific_path]
```

Comment:
Yes, my bad. I misread your code. You don't need to preserve the original routing, which will simplify a lot of the code. You can assume it follows the department/position format - hence my comment in the issue about being careful to update the hardlink on the public site when we roll this out.